### ISSUE TITLE
QNN: PyTorch v/s TF

### ISSUE THREAD

**Issue by amitkumarj441** [wabyking/qnn#2]
1.  **Can we have separate [releases](https://help.github.com/articles/creating-releases/) for QNN in TF/PyTorch?**
       For the sake of easiness in running separate versions and porting separate releases could be later packaged (via docker) to make an easy flow installation (in one go, instead of setting up dependencies).

2.  **TF & PyTorch comparison?**
       I realize these two version almost has the same network, the same loss function, the same optimizer parameters and the result might be totally different (didn't ran those versions separately yet, just TF), but would be great if we could look at these as well. I have seen that some of the model is using random initialization, we might want to switch to constant initialization, followed by seeded initialization or keep random initialization everywhere provided that we should specify the trade-offs.
       Also if any of you have isolated a specific operation that gives an incorrect or a different result (in either of the versions), it would be worth filing this issue with reproduction steps.

3. Apart from all these, we need to aggregate all the repeated dependencies at one place like numpy/pands, keras etc. in `utils.py` or `settings.py` (whatever name you like) just to avoid over-redundancy which process/train models a bit faster than earlier (especially during GPU access). This is an initial step for looking at optimizing the codes .

**Comment by wabyking** [comment]
There is another version with pytorch, but somehow messy, in case someone is interested.
https://github.com/qiuchili/qnn_torch
